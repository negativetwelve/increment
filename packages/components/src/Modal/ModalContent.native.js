// Libraries
import React from 'react';
import ReactNative from 'react-native';
import PropTypes from 'prop-types';

const ModalContent = ({isOpen, onClose, children, ...props}) => (
  <ReactNative.Modal
    animated
    transparent
    animationType={'fade'}
    visible={isOpen}
    onRequestClose={onClose}
    {...props}
  >
    <ReactNative.TouchableOpacity
      activeOpacity={1}
      onPress={onClose}
      style={{
        flex: 1,
        backgroundColor: 'rgba(74, 74, 74, 0.7)',
      }}
    />
    <ReactNative.View
      pointerEvents={'box-none'}
      style={{
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
      }}
    >
      {children}
    </ReactNative.View>
  </ReactNative.Modal>
);

// --------------------------------------------------
// Props
// --------------------------------------------------
ModalContent.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};

ModalContent.defaultProps = {
};

export default ModalContent;
