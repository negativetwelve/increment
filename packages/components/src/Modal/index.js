// Libraries
import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import styled from 'styled-components';

class ModalAdapter extends React.Component {
  state = {
    isOpen: this.props.initialIsOpen,
  };

  handleOpen = () => {
    this.setState({isOpen: true}, () => {
      if (this.props.onOpen) {
        this.props.onOpen();
      }
    });
  };

  handleClose = () => {
    this.setState({isOpen: false}, () => {
      if (this.props.onClose) {
        this.props.onClose();
      }
    });
  };

  handleRequestOpen = () => {
    // We need to open on the next event so the setState
    // isn't batched with clicking on the overlay to close.
    setTimeout(this.handleOpen, this.props.openDelay);
  };

  handleRequestClose = () => {
    this.handleClose();
  };

  render() {
    const {className, trigger, children, ...props} = this.props;
    const {isOpen} = this.state;
    const contentClassName = `${className}__content`;
    const overlayClassName = `${className}__overlay`;

    return (
      <React.Fragment>
        {trigger({isOpen, handleOpen: this.handleRequestOpen})}
        <ReactModal
          ariaHideApp={false}
          isOpen={isOpen}
          className={contentClassName}
          portalClassName={className}
          overlayClassName={overlayClassName}
          onRequestClose={this.handleRequestClose}
          {...props}>
          {children({isOpen, handleClose: this.handleRequestClose})}
        </ReactModal>
      </React.Fragment>
    );
  }
}

// --------------------------------------------------
// Props
// --------------------------------------------------
ModalAdapter.propTypes = {
  initialIsOpen: PropTypes.bool,
  openDelay: PropTypes.number,
  onOpen: PropTypes.func,
  onClose: PropTypes.func,
};

ModalAdapter.defaultProps = {
  initialIsOpen: false,
  openDelay: 200,
  onOpen: () => {},
  onClose: () => {},
};

const Modal = styled(ModalAdapter).attrs({
  suppressClassNameWarning: true,
})`
  &__overlay {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    background-color: rgba(74, 74, 74, 0.7);
  }

  &__content {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    overflow: auto;
    -webkit-overflow-scrolling: touch;
    outline: none;
    pointer-events: none;
  }

  &__content * {
    pointer-events: all;
  }
`;

export default Modal;
