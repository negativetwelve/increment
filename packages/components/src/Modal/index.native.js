// Libraries
import React from 'react';
import ReactNative from 'react-native';
import PropTypes from 'prop-types';

class Modal extends React.Component {
  state = {
    isOpen: this.props.initialIsOpen,
  };

  handleOpen = () => {
    this.setState({isOpen: true}, () => {
      if (this.props.onOpen) {
        this.props.onOpen();
      }
    });
  };

  handleClose = () => {
    this.setState({isOpen: false}, () => {
      if (this.props.onClose) {
        this.props.onClose();
      }
    });
  };

  handleRequestOpen = () => {
    // We need to open on the next event so the setState
    // isn't batched with clicking on the overlay to close.
    setTimeout(this.handleOpen, 0);
  };

  handleRequestClose = () => {
    this.handleClose();
  };

  render() {
    const {trigger, children, ...props} = this.props;
    const {isOpen} = this.state;

    return (
      <React.Fragment>
        {trigger({isOpen, handleOpen: this.handleRequestOpen})}
        <ReactNative.Modal
          animated
          transparent
          animationType={'fade'}
          visible={isOpen}
          onRequestClose={this.handleRequestClose}
          {...props}>
          <ReactNative.TouchableOpacity
            activeOpacity={1}
            onPress={this.handleRequestClose}
            style={{
              flex: 1,
              backgroundColor: 'rgba(74, 74, 74, 0.7)',
            }}
          />
          <ReactNative.View
            pointerEvents={'box-none'}
            style={{
              position: 'absolute',
              top: 0,
              left: 0,
              right: 0,
              bottom: 0,
            }}>
            {children({handleClose: this.handleRequestClose})}
          </ReactNative.View>
        </ReactNative.Modal>
      </React.Fragment>
    );
  }
}

// --------------------------------------------------
// Props
// --------------------------------------------------
Modal.propTypes = {
  initialIsOpen: PropTypes.bool,
  onOpen: PropTypes.func,
  onClose: PropTypes.func,
};

Modal.defaultProps = {
  initialIsOpen: false,
  onOpen: () => {},
  onClose: () => {},
};

export default Modal;
