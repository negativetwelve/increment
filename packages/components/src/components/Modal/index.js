// Libraries
import React from 'react';
import Modal from 'react-modal';
import styled from 'styled-components';

Modal.setAppElement('#___gatsby');

class ModalAdapter extends React.Component {
  state = {
    isOpen: false,
  };

  handleOpen = () => {
    // HACK(mark): We need to open on the next event so the setState
    // isn't batched with clicking on the overlay to close.
    setTimeout(() => this.setState({isOpen: true}), 0);
  };

  handleClose = () => {
    this.setState({isOpen: false});
  };

  handleRequestClose = () => {
    this.handleClose();
  };

  render() {
    const {className, trigger, children, ...props} = this.props;
    const {isOpen} = this.state;
    const contentClassName = `${className}__content`;
    const overlayClassName = `${className}__overlay`;

    return (
      <React.Fragment>
        {trigger({isOpen, handleOpen: this.handleOpen})}
        <Modal
          isOpen={isOpen}
          className={contentClassName}
          portalClassName={className}
          overlayClassName={overlayClassName}
          onRequestClose={this.handleRequestClose}
          {...props}>
          {children({isOpen, handleClose: this.handleClose})}
        </Modal>
      </React.Fragment>
    );
  }
}

const StyledModal = styled(ModalAdapter)`
  &__overlay {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    background-color: rgba(74, 74, 74, 0.7);
  }

  &__content {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    overflow: auto;
    -webkit-overflow-scrolling: touch;
    outline: none;
    pointer-events: none;
  }

  &__content * {
    pointer-events: all;
  }
`;

export default StyledModal;
