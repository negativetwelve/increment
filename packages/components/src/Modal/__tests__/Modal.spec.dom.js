// Libraries
import React from 'react';
import Modal from '../';

/* eslint-disable no-undef */
describe('Modal', () => {
  set('initialIsOpen', () => false);
  set('trigger', () => ({handleOpen}) => null);
  set('children', () => ({handleClose}) => null);
  set('props', () => ({
    initialIsOpen,
    trigger,
    children,
  }));

  context('with initialIsOpen: false', () => {
    set('initialIsOpen', () => false);
    itRenders(() => <Modal {...props} />);
  });

  context('with initialIsOpen: true', () => {
    set('initialIsOpen', () => true);
    itRenders(() => <Modal {...props} />);
  });
});
