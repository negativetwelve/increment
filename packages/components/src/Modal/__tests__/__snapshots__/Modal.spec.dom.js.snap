// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal with initialIsOpen: false renders snapshot 1`] = `null`;

exports[`Modal with initialIsOpen: true renders snapshot 1`] = `null`;
